core/src/test/java/uk/gov/justice/services/core/json/JsonValidationLoggerTest.java

```diff
@@ -0,0 +1,133 @@
+package uk.gov.justice.services.core.json;
+
+import static com.jayway.jsonassert.JsonAssert.with;
+import static com.jayway.jsonassert.impl.matcher.IsCollectionWithSize.hasSize;
+import static com.jayway.jsonpath.matchers.JsonPathMatchers.hasJsonPath;
+import static java.lang.String.format;
+import static java.nio.charset.Charset.defaultCharset;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static uk.gov.justice.services.core.json.JsonValidationLogger.toValidationTrace;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.everit.json.schema.Schema;
+import org.everit.json.schema.ValidationException;
+import org.everit.json.schema.loader.SchemaLoader;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+
+public class JsonValidationLoggerTest {
+
+    private String result;
+    private ValidationException validationException;
+    private final static String SCHEMA_LOCATION_PATTERN = "/json/schema/%s.json";
+    private final static String JSON_LOCATION_PATTERN = "/json/%s.json";
+
+    @Before
+    public void setup() throws IOException {
+        try {
+            schema().validate(badObject());
+            fail("Test should have resulted in validation errors");
+        } catch (ValidationException ex) {
+            validationException = ex;
+        }
+        result = toValidationTrace(validationException);
+    }
+
+    @Test
+    public void shouldHaveTopLevelMessage() throws IOException {
+        with(result)
+                .assertEquals("$.message", "#: 2 schema violations found")
+                .assertEquals("$.violation", "#");
+    }
+
+    @Test
+    public void shouldHaveTopLevelTitle() throws IOException {
+        with(result).assertEquals("$.violatedSchema", "Title of a schema for testing - ");
+    }
+
+    @Test
+    public void shouldHaveSubLevelTitle() throws IOException {
+        with(result)
+                .assertEquals("$.causingExceptions[1].violatedSchema", "Title of a schema for testing - ");
+    }
+
+    @Test
+    public void shouldHaveDescription() throws IOException {
+        with(result)
+                .assertEquals("$.causingExceptions[0].violatedSchema",
+                        "ingredients - List ingredients and quantities for recipe");
+    }
+
+    @Test
+    public void shouldSpotMissingQuantity() throws IOException {
+        with(result)
+                .assertEquals("$.causingExceptions[0].causingExceptions[0]causingExceptions[0].message",
```

Comment:
Are these paths correct? There are a few in these tests (here and below) that have no dot between array field names